File: frontend/src/models/data/EndTreeNode.ts
import TreeNode from "./TreeNode"

export default class EndTreeNode extends TreeNode {
    value: string;

    constructor(value: string, key: string) {
        super(key);
        this.value = value;
    };

    setValue(newVal: string) {
        this.value = newVal;
    }
}

File: backend/Models/EndTreeNode.cs
using Newtonsoft.Json.Linq;

namespace Graphic_Neural_Network.backend.Models
{
    public class EndTreeNode : TreeNode
    {
        public override string Type => "endtreenode";

        public string Value { get; set; }

        public EndTreeNode(string value, string key) : base(key)
        {
            Value = value;
        }

       

        public override JObject ToJson()
        {
            var jo = base.ToJson();
            jo["value"] = this.Value;
            return jo;
        }

        public void SetValue(string newVal)
        {
            Value = newVal;
        }
    }

}
